src/auth/__tests__/useProfile.test.tsx

```diff
@@ -0,0 +1,134 @@
+
+type Status = 'loading' | 'error' | 'loaded';
+type DataGetters = {
+  getInfo: () => Status;
+  getProfile: () => Profile | null;
+  hasCalledGetUser: () => boolean;
+  getMockedUserData: () => User;
+};
+
+describe('useProfile', () => {
+  const loadingStatus: Status = 'loading';
+  const loadedStatus: Status = 'loaded';
+  const errorStatus: Status = 'error';
+  const statusIndicatorElementId = 'status-indicator';
+  const profileElementId = 'profile';
+  const noProfile = { noProfile: true };
+  const TestProfileComponent = ({
+    callCounter,
+  }: {
+    callCounter: () => number;
+  }) => {
+    const hasLoadStarted = callCounter() > 0;
+    const { profile, loading, error } = useProfile();
+    const isFinished = hasLoadStarted && loading === false;
+
+    if (error) {
+      return <span id={statusIndicatorElementId}>{errorStatus}</span>;
+    }
+    if (!isFinished) {
+      return <span id={statusIndicatorElementId}>{loadingStatus}</span>;
+    }
+    return (
+      <div>
+        <span id={statusIndicatorElementId}>{loadedStatus}</span>
+        <span id={profileElementId}>
+          {JSON.stringify(profile ? profile : noProfile)}
+        </span>
+      </div>
+    );
+  };
+  const renderTestComponent = (
+    overrides?: MockedUserOverrides,
+    error = false
+  ): DataGetters => {
+    const userData = mockUserCreator(overrides);
+    const mockedGetUser = jest.spyOn(authService, 'getUser');
+
+    if (error) {
+      mockedGetUser.mockRejectedValueOnce(null);
+    } else {
+      mockedGetUser.mockResolvedValueOnce(userData);
+    }
+    const result = render(
+      <TestProfileComponent
+        callCounter={() => mockedGetUser.mock.calls.length}
+      />
+    );
+    const { container } = result;
+    const getElementById = (id: string) =>
+      container.querySelector(`#${id}`) as HTMLElement;
+    return {
+      getInfo: () =>
+        getElementById(statusIndicatorElementId).innerHTML as Status,
+      getProfile: () => {
+        const data = getElementById(profileElementId).innerHTML;
+        return JSON.parse(data);
+      },
+      hasCalledGetUser: () => mockedGetUser.mock.calls.length > 0,
+      getMockedUserData: () => userData,
+    };
+  };
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  it('should return the profile which the authService.getUser() provides where amr is always an array', async () => {
+    const amr = 'string-arm';
+    const {
+      getInfo,
+      getProfile,
+      getMockedUserData,
+      hasCalledGetUser,
+    } = renderTestComponent({
+      profileOverrides: { amr },
+    });
+    const userData = getMockedUserData();
+    await waitFor(() => expect(getInfo()).toEqual(loadedStatus));
+    expect(hasCalledGetUser()).toBeTruthy();
+    const profileWithConvertedAmr = {
+      ...userData.profile,
+      amr: [amr],
+    };
+    expect(getProfile()).toEqual(profileWithConvertedAmr);
+  });
+
+  it('should return null if user.expired !== false', async () => {
```

Comment:
This test name could be more readable. Perhaps something like:
> should provide no profile if it has expired

Reply:
The returned value of user.expired can be "undefined", if the expires_in the token is "undefined".  That is why "user.expired !== false" is used instead of just "!user.expired". 

But the name of the test does not need to reflect that. Added another test for undefined.